doc/source/whatsnew/v2.0.0.rst

```diff
@@ -49,6 +49,7 @@ Other enhancements
 - Fix ``test`` optional_extra by adding missing test package ``pytest-asyncio`` (:issue:`48361`)
 - :func:`DataFrame.astype` exception message thrown improved to include column name when type conversion is not possible. (:issue:`47571`)
 - :meth:`DataFrame.to_json` now supports a ``mode`` keyword with supported inputs 'w' and 'a'. Defaulting to 'w', 'a' can be used when lines=True and orient='records' to append record oriented json lines to an existing json file. (:issue:`35849`)
+- :func:`to_datetime` now handles ``datetime.datetime`` and :class:`Timestamp` and applies the ``format`` argument on them instead of raising a ``ValueError``. (:issue:`49298`)
```

Comment:
I mean that the `strftime` method is used on each `datetime.datetime` and `Timestamp` object using the passed `format`. Maybe:

> `to_datetime` now formats `datetime.datetime`, `Timestamp` and `str` objects in the input to return a format-consistent output instead of raising a `ValueError`